src/main/java/net/snowflake/client/core/JsonSqlInput.java

```diff
@@ -0,0 +1,341 @@
+    } else {
+      return null;
+    }
+  }
+
+  private Timestamp getTimestampFromFormat(String format, String value) {
+    String rawFormat = (String) session.getCommonParameters().get(format);
+    if (rawFormat == null || rawFormat.isEmpty()) {
+      rawFormat = (String) session.getCommonParameters().get("TIMESTAMP_OUTPUT_FORMAT");
+    }
+    SnowflakeDateTimeFormat formatter = SnowflakeDateTimeFormat.fromSqlFormat(rawFormat);
+    return formatter.parse(value).getTimestamp();
+  }
+
+  @Override
+  public Reader readCharacterStream() throws SQLException {
+    throw new SnowflakeLoggedFeatureNotSupportedException(session, "readCharacterStream");
+  }
+
+  @Override
+  public InputStream readAsciiStream() throws SQLException {
+    throw new SnowflakeLoggedFeatureNotSupportedException(session, "readAsciiStream");
+  }
+
+  @Override
+  public InputStream readBinaryStream() throws SQLException {
+    throw new SnowflakeLoggedFeatureNotSupportedException(session, "readBinaryStream");
+  }
+
+  @Override
+  public Object readObject() throws SQLException {
+    // TODO structuredType return map
+    throw new SnowflakeLoggedFeatureNotSupportedException(session, "readCharacterStream");
+  }
+
+  @Override
+  public Ref readRef() throws SQLException {
+    throw new SnowflakeLoggedFeatureNotSupportedException(session, "readRef");
+  }
+
+  @Override
+  public Blob readBlob() throws SQLException {
+    throw new SnowflakeLoggedFeatureNotSupportedException(session, "readBlob");
+  }
+
+  @Override
+  public Clob readClob() throws SQLException {
+    throw new SnowflakeLoggedFeatureNotSupportedException(session, "readClob");
+  }
+
+  @Override
+  public Array readArray() throws SQLException {
+    throw new SnowflakeLoggedFeatureNotSupportedException(session, "readArray");
+  }
+
+  @Override
+  public boolean wasNull() throws SQLException {
+    return false; // nulls are not allowed in structure types
+  }
+
+  @Override
+  public URL readURL() throws SQLException {
+    throw new SnowflakeLoggedFeatureNotSupportedException(session, "readCharacterStream");
+  }
+
+  @Override
+  public NClob readNClob() throws SQLException {
+    throw new SnowflakeLoggedFeatureNotSupportedException(session, "readNClob");
+  }
+
+  @Override
+  public String readNString() throws SQLException {
+    throw new SnowflakeLoggedFeatureNotSupportedException(session, "readNString");
+  }
+
+  @Override
+  public SQLXML readSQLXML() throws SQLException {
+    throw new SnowflakeLoggedFeatureNotSupportedException(session, "readSQLXML");
+  }
+
+  @Override
+  public RowId readRowId() throws SQLException {
+    throw new SnowflakeLoggedFeatureNotSupportedException(session, "readRowId");
+  }
+
+  @Override
+  public <T> T readObject(Class<T> type) throws SQLException {
+    return withNextValue(
+        (__, jsonNode, fieldMetadata) -> {
+          // TODO structuredType what if it is not an object but i.e. string?
```

Comment:
where will it be implemented?

Reply:
I removed it. In response we got type object in this case.